lightning/src/ln/onion_utils.rs

```diff
@@ -425,12 +425,24 @@ pub(super) fn construct_onion_packet(
 #[allow(unused)]
 pub(super) fn construct_trampoline_onion_packet(
 	payloads: Vec<msgs::OutboundTrampolinePayload>, onion_keys: Vec<OnionKeys>,
-	prng_seed: [u8; 32], associated_data: &PaymentHash, length: u16,
+	prng_seed: [u8; 32], associated_data: &PaymentHash, length: Option<u16>,
 ) -> Result<msgs::TrampolineOnionPacket, ()> {
-	let mut packet_data = vec![0u8; length as usize];
+	let minimum_packet_length: usize = payloads.iter().map(|p| p.serialized_length() + 32).sum();
 
+	assert!(
+		minimum_packet_length < ONION_DATA_LEN,
+		"Trampoline onion packet must be smaller than outer onion"
+	);
+
+	let packet_length = length.unwrap_or(minimum_packet_length as u16) as usize;
```

Comment:
```suggestion
	let packet_length = length.map(|l| usize::from(l)).unwrap_or(minimum_packet_length);
```

`usize::from` guarantees no information loss, `as usize` does not.